### PR TITLE
[Refactor] BugActionMessageCreator using ActionMessageDispatcher

### PR DESCRIPTION
**Pull request by smoralesd**
#### Description of changes

 Simple refactor to update BugActionMessageCreator to use ActionMessageDispatcher (instead of inherit from BaseActionMessageCreator)

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Added relevant unit test for your changes. (`npm run test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`npm run precheckin`)
- [ ] Added screenshots/GIFs for UI changes.

### CODE REVIEW

File: src/common/message-creators/bug-action-message-creator.ts [hunk 1]
```diff
@@ -1,37 +1,32 @@
...
 import { FILE_ISSUE_CLICK, FileIssueClickService, TelemetryEventSource } from '../telemetry-events';
-import { BaseActionMessageCreator } from './base-action-message-creator';
+import { ActionMessageDispatcher } from './action-message-dispatcher';
 
-export class BugActionMessageCreator extends BaseActionMessageCreator {
+export class BugActionMessageCreator {
     constructor(
-        postMessage: (message: Message) => void,
-        tabId: number,
+        private readonly dispatcher: ActionMessageDispatcher,
```
Comment (peterdur) [inline src/common/message-creators/bug-action-message-creator.ts]: Not blocking: we could make the other constructor parameters here `readonly` too, yes?

File: src/tests/unit/tests/common/message-creators/bug-action-message-creator.test.ts [hunk 2]
```diff
@@ -1,77 +1,70 @@
...
+            dispatcher.dispatchMessage({
+                messageType: Messages.SettingsPanel.OpenPanel,
+                payload: {
+                    telemetry,
+                },
+            }),
+        );
 
-        testSubject.openSettingsPanel(event);
+        testSubject.openSettingsPanel(eventStub);
```
Comment (peterdur) [inline src/tests/unit/tests/common/message-creators/bug-action-message-creator.test.ts]: If I comment out this line, the test still passes. Need a `.verifiable(Times.once())` on the mock setup.

Comment (peterdur) [reply src/tests/unit/tests/common/message-creators/bug-action-message-creator.test.ts]: (The `telemetryFactoryMock` is also not verified, but I can see that as an implementation detail of the test and not part of the contract being examined.)

File: src/tests/unit/tests/common/message-creators/bug-action-message-creator.test.ts [hunk 3]
```diff
@@ -1,77 +1,70 @@
...
-            payload,
-        };
+    it('dispatch message for trackFileIssueClick', () => {
+        const testService: FileIssueClickService = 'test file issue service' as FileIssueClickService;
+        const telemetry = { ...telemetryStub, service: testService };
+
+        telemetryFactoryMock.setup(factory => factory.forFileIssueClick(eventStub, source, testService)).returns(() => telemetry);
 
-        postMessageMock.setup(pm => pm(It.isValue(expectedMessage))).verifiable();
+        dispatcherMock.setup(dispatcher => dispatcher.sendTelemetry(FILE_ISSUE_CLICK, telemetry));
```
Comment (peterdur) [inline src/tests/unit/tests/common/message-creators/bug-action-message-creator.test.ts]: (Same note here as on line 53 above)

File: src/common/message-creators/bug-action-message-creator.ts (hunk 1 above)
Comment (smoralesd) [reply src/common/message-creators/bug-action-message-creator.ts]: Make sense. Updated.

File: src/tests/unit/tests/common/message-creators/bug-action-message-creator.test.ts (hunk 2 above)
Comment (smoralesd) [reply src/tests/unit/tests/common/message-creators/bug-action-message-creator.test.ts]: Good catch on the missing `verifiable`
As per the `telemetryFactoryMock`, in general, use of `verifiable` is not needed when the return value of the mock alter the result of the tested function. In this case, verifying the dispatcher is proof enough the telemetry factory was used indeed.

File: src/tests/unit/tests/common/message-creators/bug-action-message-creator.test.ts (hunk 3 above)
Comment (smoralesd) [reply src/tests/unit/tests/common/message-creators/bug-action-message-creator.test.ts]: Updated

**Pull request merged**